### PR TITLE
#165273518: Setup redux and create appropriate folder structure

### PR DESCRIPTION
**Pull request by bisonlou**
**What does this PR do?**

- Install redux, react-redux, and redux-thunk.
- Create an appropriate folder structure for working with redux.

**Description of Task to be completed?**

- Prepare the project for redux by installing redux, react-redux and redux-thunk.

**How should this be manually tested?**

- Install [redux dev tools](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en).
- Run the app using `npm install`.
- Run the app using `npm start`.
- Right click on the page displayed on your browser, and select `Redux devTools`.
- On the displayed tool, select the state tab, and then select the raw tab.
- You should see an empty state object.

**Any background context you want to provide?**

- The app requires redux to manage state.
- The console shows one error, and this is because the combined reducer is still empty.

**What are the relevant pivotal tracker stories?**

[#165273518](https://www.pivotaltracker.com/n/projects/2326197).

**Screenshots (if appropriate)**

image

### CODE REVIEW

**Review by llwasampijja — APPROVED** [review]
@bisonlou nice implementation on the redux setup. I checked the file structure, as well as naming conventions and they are fine!

**Review by kalsmic — APPROVED** [review]
well done, @bisonlou, with this redux workflow setup we will be able to implement our redux workflow smoothly.

**Review by JamesMudidi — APPROVED** [review]
@bisonlou Nice work here. The work is working well.
I appreciate the folder structure.

**Review by nadralia — COMMENTED** [review]
Nice work @bisonlou but window.devToolsExtension is deprecated in favour of window.__REDUX_DEVTOOLS_EXTENSION__ and window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__.  so do we still need to use it?

**Review by ZackAtama — APPROVED** [review]
Good work @bisonlou

File: src/store/store.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import thunk from 'redux-thunk';
+import { createStore, applyMiddleware, compose } from 'redux';
+import reducer from './rootReducer';
+
+const enhancers = compose(
+  applyMiddleware(thunk),
+  window.devToolsExtension && window.devToolsExtension(),
```
Comment (nadralia) [inline src/store/store.js]: +  `window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()`

File: src/store/store.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import thunk from 'redux-thunk';
+import { createStore, applyMiddleware, compose } from 'redux';
+import reducer from './rootReducer';
+
```
Comment (kcharles52) [inline src/store/store.js:5]:
```suggestion
import { composeWithDevTools } from "redux-devtools-extension";
```

File: src/store/store.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+const enhancers = compose(
```
Comment (kcharles52) [inline src/store/store.js]:
```suggestion
const enhancers = composeWithDevTools(applyMiddleware(thunk))
```

File: src/store/store.js (hunk 1 above)
Comment (kcharles52) [inline src/store/store.js]:
```suggestion

```

**Review by kcharles52 — CHANGES_REQUESTED** [review]

File: src/store/store.js (hunk 1 above)
Comment (bisonlou) [reply src/store/store.js]: Duly noted @nadralia I will be using `redux-devtools-extension`. Thanks.

File: src/store/store.js (hunk 2 above)
Comment (bisonlou) [reply src/store/store.js:5]: Duly noted @kcharles52. I will be using `composeWithDevTools`

**Review by kcharles52 — APPROVED** [review]

**Pull request merged**